File: TypeScript/1493. Longest Subarray of 1's After Deleting One Element.ts
// noinspection JSUnusedGlobalSymbols
const longestSubarray = (nums: number[]): number => {
  let left = 0
  let right = 0

  for (let remove = 1; right < nums.length; right++) {
    if (nums[right] == 0) {
      remove--
    }

    if (remove < 0 && nums[left++] == 0) {
      remove++
    }
  }

  return right - left - 1
}

File: C-Sharp/1493. Longest Subarray of 1's After Deleting One Element.cs
using System.Diagnostics.CodeAnalysis;

namespace LeetCode.C_Sharp;

[SuppressMessage("ReSharper", "UnusedType.Global")]
[SuppressMessage("ReSharper", "UnusedMember.Global")]
public partial class Solution
{
    public static int LongestSubarray(int[] nums)
    {
        int left = 0;
        int right = 0;

        for (int remove = 1; right < nums.Length; right++)
        {
            if (nums[right] == 0)
            {
                remove--;
            }

            if (remove < 0 && nums[left++] == 0)
            {
                remove++;
            }
        }

        return right - left - 1;
    }
}
